rs/sns/governance/src/lib.rs

```diff
@@ -121,12 +126,35 @@ fn validate_required_field<'a, Inner>(
         .ok_or_else(|| format!("The {} field must be populated.", field_name))
 }
 
-/// Return an Err whose inner value describes (in detail) what is wrong with a
-/// field value, and where within some (Protocol Buffers message) struct.
-fn field_err(field_name: &str, field_value: impl Debug, defect: &str) -> Result<(), String> {
+/// Return an Err whose inner value describes (in detail) what is wrong with a field value (should
+/// be bounded), and where within some (Protocol Buffers message) struct.
+///
+/// Only up to the first `MAX_SCALAR_FIELD_LEN_BYTES` bytes will be taken from `field_value`.
+fn field_err(field_name: &str, field_value: String, defect: &str) -> Result<(), String> {
+    let mut bounded_field_value = String::new();
+    // Searching for the longest byte prefix that is a valid string is expensive in the worst case.
+    // This is because unicode characters are unbounded. Instead, concatenate characters one-by-one,
```

Comment:
> unicode characters are unbounded

Whaa??

Reply:
Okay, this comment is misleading. Actually there are ligatures of unbounded size, but they would be represented as multiple `char`s in Rust's utf-8 encoding. I'll adjust the comment.